src/HookCoveredCallImplV1.sol

```diff
@@ -292,16 +302,23 @@ contract HookCoveredCallImplV1 is
     uint128 strikePrice,
     uint32 expirationTime
   ) external nonReentrant whenNotPaused returns (uint256) {
+    address msgSender = msg.sender;
+    emit log_named_address("msg.sender", msgSender);
+    if (msg.sender == execAddr) {
```

Comment:
this should be a modifier, that way we can always just use the msgSender variable. Or an alternative would be a library or internal function somehow, and we chang all other usages to msgSender (which falls back to msg.Sender if not the trusted proxy address)